packages/gateway/src/services/ethereumClient.ts

```diff
@@ -47,12 +48,30 @@ export class EthereumClient<chain extends Chain> {
     } catch {
       /** error is expected when it isn't a contract */
     }
+
     return owner
   }
 
+  async getNFTOwner(node: Hex): Promise<Hex> {
+    try {
+      return await this.client.readContract({
+        address: this.registrarAddress,
+        abi: [
+          parseAbiItem('function ownerOf(uint256 id) view returns (address)'),
+        ],
+        functionName: 'ownerOf',
+        args: [fromHex(node, 'bigint')],
+      })
+    } catch {
+      return zeroAddress
+    }
+  }
+
   async verifyOwnership(node: Hex, address: Hex): Promise<boolean> {
-    if (!this.registryAddress) return false
-    return (await this.getOwner(node)) === address
+    return (
+      (await this.getOwner(node)) === address ||
```

Comment:
It might be helpful to explicitly clarify that this parameter is the `ownerAddress`, and not any other type of address

Reply:
it isn't yet the owner at this point